static/src/javascripts/projects/common/modules/support/articleCount.ts

```diff
@@ -0,0 +1,87 @@
+import { storage } from '@guardian/libs';
+import {
+	getWeeklyArticleHistory,
+	incrementWeeklyArticleCount,
+} from '@guardian/support-dotcom-components';
+import type { WeeklyArticleHistory } from '@guardian/support-dotcom-components/dist/shared/src/types/targeting';
+import { storageKeyDailyArticleCount } from 'common/modules/onward/history';
+import { getArticleCountConsent } from 'common/modules/support/supportMessaging';
+
+export interface DailyArticle {
+	day: number;
+	count: number;
+}
+
+type DailyArticleHistory = DailyArticle[];
+
+const today = Math.floor(Date.now() / 86400000); // 1 day in ms
+
+const getDailyArticleHistory = (): DailyArticleHistory | undefined => {
+	// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment -- reading from local storage
+	const item = storage.local.get(storageKeyDailyArticleCount);
+	if (item) {
```

Comment:
We could use a [type predicate](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) here?

Reply:
I've pushed up a commit defining `isDailyArticleHistory`, which just checks if it's an array (anything more seems excessive?)